src/codegen/type/varchar_type.cpp

```diff
@@ -212,6 +212,97 @@ struct DateTrunc : public TypeSystem::BinaryOperatorHandleNull {
   }
 };
 
+struct Trim : public TypeSystem::UnaryOperator {
+  bool SupportsType(const Type &type) const override {
+    return type.GetSqlType() == Varchar::Instance();
+  }
+
+  Type ResultType(UNUSED_ATTRIBUTE const Type &val_type) const override {
+    return Varchar::Instance();
+  }
+
+  Value Eval(CodeGen &codegen, const Value &val) const override {
+    llvm::Value *ret = codegen.Call(StringFunctionsProxy::Trim,
+                                    {val.GetValue(), val.GetLength()});
+
+    llvm::Value *str_ptr = codegen->CreateExtractValue(ret, 0);
+    llvm::Value *str_len = codegen->CreateExtractValue(ret, 1);
+    return Value(Varchar::Instance(), str_ptr, str_len);
+  }
+};
+
+struct BTrim : public TypeSystem::BinaryOperator {
```

Comment:
Subclass `BinaryOperatorHandleNull` in order to properly handle `NULL` inputs. This should be the case for all functions you implement here.